lib/elastic_apm/normalizers/rails/active_record.rb

```diff
@@ -17,7 +17,14 @@ class SqlNormalizer < Normalizer
         def initialize(*args)
           super
 
-          @summarizer = SqlSummarizer.new
+          @summarizer ||=
+            if ElasticAPM.agent&.config&.use_experimental_sql_parsing
+              require 'elastic_apm/sql/signature'
+              Sql::Signature::Summarizer.new
+            else
+              SqlSummarizer.new
+            end
```

Comment:
Another way to do this is to have a module method `Sql.summarizer` that returns the summarizer object and do this config-checking logic there. That way, you only have to update the code in one place when you make the switch to the new implementation.

Reply:
Good idea, did that 👍